components/integrations/edit_oauth_app/edit_oauth_app.tsx

```diff
@@ -66,15 +54,15 @@ export default class EditOAuthApp extends React.PureComponent {
         }
     }
 
-    editOAuthApp = async (app) => {
+    editOAuthApp = async (app: OAuthApp) => {
         this.newApp = app;
 
         if (this.props.oauthApp.id) {
             app.id = this.props.oauthApp.id;
         }
 
-        if (this.props.oauthApp.token) {
-            app.token = this.props.oauthApp.token;
+        if (this.props.oauthApp.client_secret) {
```

Comment:
Since there's no `OAuthApp.token` field, I think we can just remove this block entirely instead of pointing this code at a different field.